timesketch/api/v1/resources.py

```diff
@@ -408,6 +467,291 @@ def post(self):
+                indices = query_filter.get('indices', sketch_indices)
+                if not indices or '_all' in indices:
+                    indices = [
+                        t.searchindex.index_name
+                        for t in sketch.timelines
+                    ]
+
+                query_dsl = view.query_dsl
+                if query_dsl:
+                    query_dict = json.loads(query_dsl)
+                    if not query_dict:
+                        query_dsl = None
+
+                # TODO (kiddi): Include all fields, not just the default one.
+                # TODO (kiddi): Enable scrolling support.
+                result = self.datastore.search(
+                    sketch_id=sketch.id,
+                    query_string=view.query_string,
+                    query_filter=query_filter,
+                    query_dsl=query_dsl,
+                    indices=indices)
+
+                fh = io.StringIO()
+                lines = []
+                for event in result['hits']['hits']:
+                    line = event['_source']
+                    line.setdefault('label', [])
+                    line['_id'] = event['_id']
+                    line['_type'] = event['_type']
+                    line['_index'] = event['_index']
+                    try:
+                        for label in line['timesketch_label']:
+                            if sketch.id != label['sketch_id']:
+                                continue
+                            line['label'].append(label['name'])
+                        del line['timesketch_label']
+                    except KeyError:
+                        pass
+
+                    lines.append(line)
+                data_frame = pd.DataFrame(lines)
+                del lines
+                data_frame.to_csv(fh, index=False)
+                fh.seek(0)
+                zip_file.writestr(
+                    'views/{0:s}.csv'.format(name), data=fh.read())
+
+                if not view.user:
+                    username = 'System'
+                else:
+                    username = view.user.username
+                meta = {
+                    'name': view.name,
+                    'view_id': view.id,
+                    'description': view.description,
+                    'query_string': view.query_string,
+                    'query_filter': view.query_filter,
+                    'query_dsl': view.query_dsl,
+                    'username': username,
+                    'sketch_id': view.sketch_id,
+                }
+                zip_file.writestr(
+                    'views/{0:s}.meta'.format(name), data=json.dumps(meta))
+
+            # TODO (kiddi): Add in aggregation group support.
+            # TODO (kiddi): Add in support for comments/stars/labels.
+            # TODO (kiddi): Add in support for all tagged events (includes
+            # a metadata file with a list of all tags and their count).
+
+        if is_archived:
```

Comment:
should be `if not`, right?

Reply:
no.. in the beginning there is a check to see if the `archive` label is applied. 

If it is, then this variable `is_archived` is set... this means that at the beginning of this phase `is_archived` is set to True... that means in order to do the export I needed to `unarchive` and therefore I need to `archive` again now....

whether this variable should be named something more "clear" is another question.... (renamed, is this more clear?)

Reply:
ah, missed that one. Cool, all good.